lib/gollum.rb

```diff
@@ -33,4 +33,25 @@ def initialize(dir, existing, attempted, message = nil)
       super(message || "Cannot write #{@dir}/#{@attempted_path}, found #{@dir}/#{@existing_path}.")
     end
   end
+  
+  class TemplateFilter
+    @@filters = {}
+
+    def self.filters
+      @@filters
+    end
+
+    def self.add_filter(pattern, &replacement)
+      @@filters[pattern] = replacement
+    end
+
+    def self.apply_filters(data)
+      template = data
+      @@filters.each do |pattern, replacement|
+        template = template.gsub(pattern, replacement.call)
```

Comment:
Note that `gsub!` would not work because `gsub!` returns `nil` when the call does not succeed (which could happen if a filter is defined but not included in `_Template`).